src/lib.rs

```diff
@@ -13,25 +13,31 @@
 //! Each module has an `errors.rs` defining module specific errors that are used
 //! within the crate. This exposes some of the
 //! module errors that are publicly relevant.
-//! All errors implement `Debug`, `Display`, `PartialEq`, and `description` of
-//! the `Error` trait.
+//! All errors implement the [`Error`](`std::error::Error`) trait and
+//! [`PartialEq`](`std::cmp::PartialEq`).
 //!
-//! The C FFI API exposes the errors represented  as u16.
+//! The high-level errors API in [`error`](`error`) are a different error
+//! representation as `u16` for C FFI APIs.
+#![cfg_attr(docsrs, feature(doc_cfg))]
+#![forbid(unsafe_code)]
 
 #[macro_use]
 mod utils;
 
+#[macro_use]
+pub mod error;
+
 mod ciphersuite;
 mod codec;
-mod config;
+pub mod config;
```

Comment:
Why do we make this public here and not through `prelude.rs`?

Reply:
The prelude is only supposed to re-export a subset of the public API that's most commonly used to make usage simpler. Most modules here will become public again to expose their API. I only touched the ones necessary for now (docs won't work otherwise for example as you'd have to point to the prelude, which is not what you want).